packages/typespec-ts/src/modular/buildCodeModel.ts

```diff
@@ -1644,7 +1662,7 @@ export function emitCodeModel(
   simpleTypesMap.clear();
   const allModels = getAllModels(dpgContext);
   for (const model of allModels) {
-    getType(dpgContext, model);
+    getType(dpgContext, model.__raw!);
```

Comment:
curious why there's no breaking in getType ?

Reply:
see comments https://github.com/Azure/typespec-azure/pull/3547#discussion_r1329752601